packages/client/src/clients/in-app-message/types.ts

```diff
@@ -0,0 +1,92 @@
+import { GenericData, PageInfo } from "@knocklabs/types";
+
+import { NetworkStatus } from "../../networkStatus";
+import { NotificationSource } from "../messages/interfaces";
+
+export interface CommonContentField {
+  key: string;
+  label: string;
+  rendered: string;
+  type: "text" | "markdown" | "boolean" | "textarea";
+  value: string;
+}
+
+export interface ButtonContentField {
+  key: string;
+  type: "button";
+  text: Exclude<CommonContentField, "key | type"> & {
+    key: "text";
+    type: "text";
+  };
+  action: Exclude<CommonContentField, "key | type"> & {
+    key: "action";
+    type: "text";
+  };
+}
+
+export type ContentField = CommonContentField | ButtonContentField;
```

Comment:
I wonder if these should be prefixed with "InAppMessage" also. Not a big deal if they aren't exposed outside this package though.

Reply:
They are exported, so I'll add the prefix!